apis/vshn/v1/common_types.go

```diff
@@ -184,3 +187,24 @@ type Security struct {
 	// AllowedNamespaces defines a list of namespaces from where the service can be reached in the claim namespace
 	AllowedNamespaces []string `json:"allowedNamespaces,omitempty"`
 }
+
+type VSHNAccess struct {
+	// User specifies the username. If all other fields are left empty
+	// then a new database with the same name and all permissions will be created.
+	// +kubebuilder:validation:Required
+	User *string `json:"user,omitempty"`
+
+	// Database is the name of the database to create, defaults to user.
+	Database *string `json:"database,omitempty"`
+
+	// Privileges specifies the privileges to grant the user. Please check
+	// the database's docs for available privileges.
+	Privileges []string `json:"privileges,omitempty"`
+
+	// WriteConnectionSecretToReference specifies the namespace and name of a
+	// Secret to which any connection details for this user should
+	// be written.
+	// If not specified, a secret with the name $claimname-$username will be
```

Comment:
Can we do that for the main secret please? :D

Reply:
Not sure, as the main `writeConnectionSecretToReference` is controlled by Crossplane.

I found out that we can't actually access the `writeConnectionSecretToReference` of the claim. We can only access the field as set in the composite. Which is distinct from the one in the claim and points to the designated `writeConnectionSecretsToNamespace` as specified in the composition. We have 0 control over that from within the comp functions.

That's why it defaults to `$claimname-$username` and not to `$mainsecretname-$username`, because there's no easy way to get that value...